### PR TITLE
Refactor tests under `sshutils` package

### PR DESCRIPTION
**Pull request by russjones**
Refactored all tests under `lib/sshutils` to use testify instead of gocheck.

### CODE REVIEW

File: lib/sshutils/server_test.go [hunk 1]
```diff
@@ -25,74 +25,66 @@ import (
...
 	_, err = clt.NewSession()
-	c.Assert(err, check.NotNil)
+	require.NotNil(t, err)
 
-	c.Assert(srv.Close(), check.IsNil)
-	wait(c, srv)
-	c.Assert(called, check.Equals, true)
+	require.NoError(t, srv.Close())
+	wait(t, srv)
+	require.Equal(t, called, true)
```
Comment (zmb3) [inline lib/sshutils/server_test.go:76]:
```suggestion
	require.True(t, called)
```

File: lib/sshutils/server_test.go [hunk 2]
```diff
@@ -103,93 +95,99 @@ func (s *ServerSuite) TestShutdown(c *check.C) {
...
 	// call new session to initiate opening new channel
 	_, err = clt.NewSession()
-	c.Assert(err, check.IsNil)
+	require.NoError(t, err)
 
 	// context will timeout because there is a connection around
 	ctx, ctxc := context.WithTimeout(context.TODO(), 50*time.Millisecond)
 	defer ctxc()
-	c.Assert(trace.IsConnectionProblem(srv.Shutdown(ctx)), check.Equals, true)
+	require.Equal(t, trace.IsConnectionProblem(srv.Shutdown(ctx)), true)
```
Comment (zmb3) [inline lib/sshutils/server_test.go:120]:
```suggestion
	require.True(t, trace.IsConnectionProblem(srv.Shutdown(ctx))
```

File: lib/sshutils/server_test.go [hunk 3]
```diff
@@ -232,19 +230,19 @@ func (s *ServerSuite) TestHostSignerFIPS(c *check.C) {
...
 		)
-		c.Assert(err != nil, check.Equals, tt.outError)
+		require.Equal(t, err != nil, tt.outError)
 	}
 }
 
-func wait(c *check.C, srv *Server) {
+func wait(t *testing.T, srv *Server) {
 	s := make(chan struct{})
 	go func() {
```
[the remark this answers is not in the document]
Comment (russjones) [reply lib/sshutils/server_test.go]: Unfortunately `srv.Wait()` does not return an error, so no way to know the timeout hit.

File: lib/sshutils/server_test.go [hunk 4]
```diff
@@ -25,74 +25,89 @@ import (
...
+		go func() {
+			srv.Wait(context.TODO())
+			s <- struct{}{}
+		}()
+
+		select {
+		case <-time.After(time.Second):
+			require.Equal(t, false, true, "exceeded waiting timeout")
+		case <-s:
+		}
```
Comment (zmb3) [inline lib/sshutils/server_test.go:79]:
```suggestion
		ctx, cancel := context.WithTimeout(context.Background(), 1 * time.Second)
		defer cancel()
		srv.Wait(ctx)
		require.NoError(t, ctx.Err())
```

File: lib/sshutils/server_test.go [hunk 5]
```diff
@@ -25,74 +25,82 @@ import (
...
-func (s *ServerSuite) TestStartStop(c *check.C) {
 	called := false
 	fn := NewChanHandlerFunc(func(_ context.Context, _ *ConnectionContext, nch ssh.NewChannel) {
 		called = true
-		err := nch.Reject(ssh.Prohibited, "nothing to see here")
-		c.Assert(err, check.IsNil)
+
+		if err := nch.Reject(ssh.Prohibited, "nothing to see here"); err != nil {
+			logrus.Errorf("Failed to reject connection: %v", err)
+		}
```
Comment (espadolini) [inline lib/sshutils/server_test.go:53]:
```suggestion
		err := nch.Reject(ssh.Prohibited, "nothing to see here")
		assert.NoError(t, err)
```

from `github.com/stretchr/testify/assert`

File: lib/sshutils/server_test.go [hunk 6]
```diff
@@ -25,74 +25,66 @@ import (
...
 	// Call new session to initiate opening new channel. This should get
 	// rejected and fail.
 	_, err = clt.NewSession()
-	c.Assert(err, check.NotNil)
+	require.NotNil(t, err)
 
-	c.Assert(srv.Close(), check.IsNil)
-	wait(c, srv)
-	c.Assert(called, check.Equals, true)
+	require.NoError(t, srv.Close())
```
[the remark this answers is not in the document]
Comment (espadolini) [reply lib/sshutils/server_test.go:90]: Seconded

File: lib/sshutils/server_test.go [hunk 7]
```diff
@@ -103,149 +111,148 @@ func (s *ServerSuite) TestShutdown(c *check.C) {
...
+
+	_, signer, err := utils.CreateCertificate("foo", ssh.HostCert)
+	require.NoError(t, err)
+
 	fn := NewChanHandlerFunc(func(_ context.Context, _ *ConnectionContext, nch ssh.NewChannel) {
-		err := nch.Reject(ssh.Prohibited, "nothing to see here")
-		c.Assert(err, check.IsNil)
+		if err := nch.Reject(ssh.Prohibited, "nothing to see here"); err != nil {
+			logrus.Errorf("Failed to reject connection: %v", err)
+		}
```
Comment (espadolini) [inline lib/sshutils/server_test.go:159]: same as above
```suggestion
		err := nch.Reject(ssh.Prohibited, "nothing to see here")
		assert.NoError(t, err)
```

File: lib/sshutils/server_test.go [hunk 8]
```diff
@@ -103,149 +111,148 @@ func (s *ServerSuite) TestShutdown(c *check.C) {
...
 		Config: ssh.Config{
 			Ciphers: []string{"aes256-ctr"},
 		},
 		Auth:            []ssh.AuthMethod{ssh.Password("abc123")},
-		HostKeyCallback: ssh.FixedHostKey(s.signer.PublicKey()),
+		HostKeyCallback: ssh.FixedHostKey(signer.PublicKey()),
 	}
 	_, err = ssh.Dial("tcp", srv.Addr(), &cc)
-	c.Assert(err, check.NotNil, check.Commentf("cipher mismatch, should fail, got nil"))
+	require.NotNilf(t, err, "cipher mismatch, should fail, got nil")
```
Comment (espadolini) [inline lib/sshutils/server_test.go:183]:
```suggestion
	require.NoError(t, err, "cipher mismatch, should fail, got nil")
```

**Review by espadolini — APPROVED** [review]

File: lib/sshutils/server_test.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (russjones) [reply lib/sshutils/server_test.go:90]: Done.

**Review by rosstimothy — APPROVED** [review]

**Pull request merged**